### ISSUE TITLE
[Bug]: PWA icons do not load on iOS/iPadOS

### ISSUE THREAD

**Issue by umdstu** [coder/code-server#4994]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Safari
- Local OS: iOS/iPadOS
- Remote OS: raspberryPi OS
- Remote Architecture: armv7l
- `code-server --version`: 4.1.0 9e620e90f53fb91338a2ba1aaa2e556d42ae52d5 with Code 1.63.0

### Steps to Reproduce

1. Open `code-server` on an iPad or iPhone
2. Add Shortcut to Homescreen (save as PWA)

### Expected

The PWA icon should be rendered as the icon when saved to the homescreen on an iPad or iPhone.

### Actual

A screen grab is the icon image instead of the PWA icon

### Does this issue happen in VS Code?

- [X] I cannot reproduce this in VS Code.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

The issue appears to be due to permissions of certain directories. If you access code-server without being logged in, and attempt to bookmark the page, you get the correct icon, which fetches and renders https://chases.io/_static/src/browser/media/pwa-icon-512.png . However, once you're logged in, the icon it tries to fetch is https://chases.io/static/resources/server/pwa-icon-512.png . Safari seems to attempt to fetch the icon unauthenticated, so it gets a 401 for that file. I manually replaced the path being used when authenticated (for the latter icon) in `workbench.html` with the path being used in `login.html` (the former path), and it loaded just fine. 

I believe that the only solution is to ensure the PWA icon path, regardless of the `.html` file used, is accessible when unauthenticated. This applies to both the path of the file, the path in the manifest.json, and anywhere else, e.g. `remoteExtensionHostAgent.js`.

**Comment by jsjoeio** [comment]
> The issue appears to be due to permissions of certain directories. If you access code-server without being logged in, and attempt to bookmark the page, you get the correct icon, which fetches and renders https://chases.io/_static/src/browser/media/pwa-icon-512.png . However, once you're logged in, the icon it tries to fetch is https://chases.io/static/resources/server/pwa-icon-512.png . Safari seems to attempt to fetch the icon unauthenticated, so it gets a 401 for that file. I manually replaced the path being used when authenticated (for the latter icon) in workbench.html with the path being used in login.html (the former path), and it loaded just fine.

> I believe that the only solution is to ensure the PWA icon path, regardless of the .html file used, is accessible when unauthenticated. This applies to both the path of the file, the path in the manifest.json, and anywhere else, e.g. remoteExtensionHostAgent.js.

I can't tell you how much we appreciate you digging into the _why_ behind this and suggesting a solution. This shouldn't be terribly difficult to do. Thank you so much!

**Comment by umdstu** [comment]
Happy to help. I was also wondering if the assets could be somehow be combined. Seems silly to have those duplicate images, unless the PWA explicitly needs it's own copy.

**Comment by jsjoeio** [comment]
> I was also wondering if the assets could be somehow be combined

There may be some historical context as to why we didn't combine them 🤔 @code-asher might know.

We're currently moving to patches (https://github.com/coder/code-server/pull/4997) so once that lands, it should be easier to help with these kinds of things.

**Comment by code-asher** [comment]
Yeah I have no idea why we were duplicating these.  We only have one copy now (under `/_static`).

**Comment by mrfoxie** [comment]
The same issue is for Android and for Windows 10-11 too

**Comment by jp-pino** [comment]
I’m still having this issue on 4.2.0 which included this fix. Maybe due to a different problem?
![IMG_0008](https://user-images.githubusercontent.com/35382610/161385072-c8cf8b4f-b056-4017-b38b-9e81a34f0d52.jpeg)

**Comment by umdstu** [comment]
Did you delete the bookmark and save it again? I haven't had a chance to test the fix BTW so can't say personally that it works for me yet.
